fdbcli/tests/metacluster_fdbcli_tests.py

```diff
@@ -42,51 +88,75 @@ def metacluster_create(cluster_file, name, tenant_id_prefix):
-    """
+def setup_tenants(management_cluster_file, data_cluster_files, tenants):
+    for tenant in tenants:
+        output = run_fdbcli_command(management_cluster_file, 'tenant create', tenant)
+        expected_output = 'The tenant `{}\' has been created'.format(tenant)
+        assert output == expected_output
 
-    parser = argparse.ArgumentParser(formatter_class=RawDescriptionHelpFormatter,
-                                     description=script_desc)
 
-    parser.add_argument('build_dir', metavar='BUILD_DIRECTORY', help='FDB build directory')
-    args = parser.parse_args()
+def configure_tenant(management_cluster_file, data_cluster_files, tenant, tenant_group=None, assigned_cluster=None):
+    command = 'tenant configure {}'.format(tenant)
+    if tenant_group:
+        command = command + ' tenant_group={}'.format(tenant_group)
+    if assigned_cluster:
+        command = command + ' assigned_cluster={}'.format(assigned_cluster)
 
-    # keep current environment variables
-    fdbcli_env = os.environ.copy()
-    cluster_files = fdbcli_env.get("FDB_CLUSTERS").split(';')
-    num_clusters = len(cluster_files)
-    assert len(cluster_files) > 1
+    output, err = run_fdbcli_command_and_get_error(management_cluster_file, command)
+    return output, err
 
-    fdbcli_bin = args.build_dir + '/bin/fdbcli'
 
+@enable_logging()
+def clear_database_and_tenants(logger, management_cluster_file, data_cluster_files):
+    subcmd1 = 'writemode on'
+    subcmd2 = 'option on SPECIAL_KEY_SPACE_ENABLE_WRITES'
+    subcmd3 = 'clearrange ' '"" \\xff'
+    subcmd4 = 'clearrange \\xff\\xff/management/tenant/map/ \\xff\\xff/management/tenant/map0'
+    output = run_fdbcli_command(management_cluster_file, subcmd1, subcmd2, subcmd3, subcmd4)
+    logger.debug(output)
+
+
+def run_tenant_test(management_cluster_file, data_cluster_files, test_func):
+    test_func(management_cluster_file, data_cluster_files)
+    clear_database_and_tenants(management_cluster_file, data_cluster_files)
+
+
+@enable_logging()
+def clusters_status_test(logger, cluster_files):
     for cf in cluster_files:
         output = metacluster_status(cf)
         assert output == "This cluster is not part of a metacluster"
 
+    num_clusters = len(cluster_files)
     names = ['meta_mgmt']
     names.extend(['data{}'.format(i) for i in range(1, num_clusters)])
-
-    tenant_id_prefix = random.randint(0, 32767)
-    metacluster_create(cluster_files[0], names[0], tenant_id_prefix)
-    for (cf, name) in zip(cluster_files[1:], names[1:]):
-        output = metacluster_register(cluster_files[0], cf, name)
+    setup_metacluster([cluster_files[0], names[0]], zip(cluster_files[1:], names[1:]))
 
     expected = """
 number of data clusters: {}
-  tenant group capacity: 0
+  tenant group capacity: {}
   allocated tenant groups: 0
 """
-    expected = expected.format(num_clusters - 1).strip()
+    expected = expected.format(num_clusters - 1, 12).strip()
```

Comment:
Where does this 12 come from? It's hard to follow the context here

Reply:
It's because there are two data clusters each of which has `max_tenant_group=6`.

Reply:
I can update the test script here if the remaining CI fails because I will have to re-trigger them anyway. :)

Reply:
I see, can this be parameterized and passed around, or at least have a comment here describing how it's calculated? This seems like it would be hard to debug if we changed one or the other since there's nothing explicitly linking the two unless you already know or walk through the entire thing once slowly

Reply:
Looks like CI has passed and this is pretty minor, so let's avoid kicking off CI again. You can either make another PR for this change, or I can apply it to my PR while I fix merge conflicts, if that's fine with you.

Reply:
Sure, I will submit another PR to update the metacluster_fdbcli_tests.py.
Thanks

Reply:
Addressed in #9406